src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1394,42 +1394,22 @@ protected void setAttribute(final short tag, final Object value) {
      * @param value the value to be checked
      * @return true if the value is valid and false otherwise
      */
+    @Deprecated
```

Comment:
I don't think there's any compelling reason to make it public.  I don't see any known subclasses of samtools using this directly.  And I think if they did they would just choose to blow up if the tag isn't valid, so that will happen automatically now.  If they really need to test it and do something else they can always catch the failure.